src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -603,9 +612,15 @@ static int client_append_common_options_in_managed_mode(
                         return r;
         }
 
-        r = dhcp6_option_append_fqdn(buf, offset, client->fqdn);
-        if (r < 0)
-                return r;
+        if (IN_SET(client->state,
+                   DHCP6_STATE_SOLICITATION,
+                   DHCP6_STATE_REQUEST,
+                   DHCP6_STATE_RENEW,
+                   DHCP6_STATE_REBIND)) {
```

Comment:
This is OK, but for simplicity, how about `if (client->state != DHCP6_STATE_STOPPING)` ?